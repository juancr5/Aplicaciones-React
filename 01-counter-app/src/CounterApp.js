import React, {useState} from 'react';
import PropTypes from 'prop-types';


/*valor por defecto en el value, en caso de ser asignado desde el index*/
const CounterApp = ({value = 0}) => {

    const [counter, setCounter]= useState(value);

    // Funcion para aumentar el contador
    const handleAdd = () => {
        setCounter(counter + 1)
        /* Otra forma de aumentar la variable
        setCounter( (c) => c+1)
        */
    }

    // Funcion para dismuniuir el contador
    const handleSubtract = () => setCounter(counter -1);

     // Funcion para dismuniuir el contador
     const handleReset = () => setCounter(value);

    return (   
        <div>
            <h1>CounterApp</h1>
         

            <h2>{counter}</h2> 

            <button onClick={handleAdd} >+1</button>
            <button onClick={handleReset} >Reset</button>
            <button onClick={handleSubtract} >-1</button>
        </div>
    );
}

// Esta linea hace obligatorio que el tipo String en la variable 
CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;