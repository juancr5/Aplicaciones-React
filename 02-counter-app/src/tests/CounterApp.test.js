import { fireEvent, render, screen } from '@testing-library/react';
import CounterApp from '../CounterApp';

describe('Pruebas en el <CounterApp />', () => {

  const initialValue = 10;

  //.toBeInTheDocument() propia de la libreria 
  test('Debe de Renderizar el Titulo en el Main', () => {
    render(<CounterApp />);
    const titleMain = screen.getByText(/CounterApp/);
    expect(titleMain).toBeInTheDocument();
  });

  test('Debe de hacer Match con el Snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de Mostrar el Valor Inicial de 100 <CounterApp value={100}>', () => {

    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();

  });

  test('Debe de Incrementar con el Botón +1', () => {

    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('11')).toBeTruthy();

  });

  test('Debe de Decrementar con el Botón -1', () => {

    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('-1'));
    expect(screen.getByText('9')).toBeTruthy();

  });

  test('Debe de Funcionar el Botón de Reset', () => {
        
    render( <CounterApp value={ 355 } /> );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    // fireEvent.click( screen.getByText('Reset') );
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    expect( screen.getByText( 355 ) ).toBeTruthy();

});


});
