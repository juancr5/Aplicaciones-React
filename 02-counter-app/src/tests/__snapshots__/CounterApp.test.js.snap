// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Debe de hacer Match con el Snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h2>
      10
    </h2>
    <button>
      +1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
    <button>
      -1
    </button>
  </div>
</div>
`;
